app/src/androidTest/java/au/com/shiftyjelly/pocketcasts/models/db/DataOpenHelper.kt

```diff
@@ -0,0 +1,689 @@
+//        db.execSQL(insert + "('ad0370b0-1118-0130-36a2-60f8472cfba6', 'Downloaded', 1, 1, 1, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 15, 0, 1, 4);");
+    }
+
+    private fun getColumnNames(db: SQLiteDatabase, tableName: String): List<String> {
+        val names: MutableList<String> = ArrayList()
+        var cursor: Cursor? = null
+        try {
+            cursor = db.rawQuery("select * from $tableName limit 1", null)
+            if (cursor != null) names.addAll(listOf(*cursor.columnNames))
+        } catch (e: Exception) {
+            Timber.e(e)
+        } finally {
+            cursor?.close()
+        }
+        return names
+    }
+
+    private fun setNullAddedDates(db: SQLiteDatabase) {
+        val cal = Calendar.getInstance()
+        cal.add(Calendar.MONTH, -1)
+        var cursor: Cursor? = null
+        try {
+            db.beginTransaction()
+            cursor = db.rawQuery("SELECT uuid FROM podcast", arrayOf())
+            if (cursor.moveToFirst()) {
+                do {
+                    val uuid = cursor.getString(0)
+                    db.execSQL("UPDATE podcast SET added_date=? WHERE uuid=?", arrayOf<Any>(cal.timeInMillis, uuid))
+                    cal.add(Calendar.MINUTE, 1)
+                } while (cursor.moveToNext())
+            }
+            db.setTransactionSuccessful()
+        } catch (e: SQLException) {
+            Timber.e(e, "Migrating database to version 2")
+        } finally {
+            try {
+                db.endTransaction()
+            } catch (exception: java.lang.Exception) {
+                Timber.e(exception)
+            }
+
+            try {
+                cursor?.close()
+            } catch (exception: java.lang.Exception) {
+                Timber.e(exception)
+            }
+        }
+    }
+
+    private fun updateLatestEpisodeFields(db: SQLiteDatabase) {
+        try {
+            with(db) {
+                beginTransaction()
+                execSQL("ALTER TABLE podcast ADD latest_episode_date INTEGER", arrayOf())
+                execSQL("UPDATE podcast SET latest_episode_uuid = NULL", arrayOf())
+                setTransactionSuccessful()
+            }
+        } catch (e: SQLException) {
+            Timber.e(e, "Migrating database to version 2")
+        } finally {
+            try {
+                db.endTransaction()
+            } catch (exception: java.lang.Exception) {
+                Timber.e(exception)
+            }
+        }
+    }
+
+    private fun clearPodcastTintColours(db: SQLiteDatabase) {
+        db.execSQL("UPDATE podcast SET primary_color = NULL, secondary_color = NULL")
+    }
+
+    @Suppress("SameParameterValue")
+    private fun maxColumn(db: SQLiteDatabase, table: String, column: String): Int {
+        val result = firstRowArray("SELECT MAX($column) FROM $table", arrayOf(), db)
+        return result?.filterNotNull()?.first()?.toInt() ?: 0
```

Comment:
`first()` throws if there are no elements.

```suggestion
        return result?.filterNotNull()?.firstOrNull()?.toInt() ?: 0
```

Or in a more concise form. This also handles cases where `toInt()` might throw.

```suggestion
        return result?.firstNotNullOfOrNull { it?.toIntOrNull() } ?: 0
```